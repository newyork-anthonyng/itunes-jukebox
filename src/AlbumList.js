import React from "react";
import PropTypes from "prop-types";

const AlbumList = ({ artistName, albums = [] }) => (
  <div>
    <h2>{artistName}</h2>
    <ul>
      {albums.map(({ id, itunesLink, albumName, albumArt }) => (
        <li key={id}>
          <a href={itunesLink} title={`See ${albumName} on iTunes`}>
            <img src={albumArt} alt={albumName} />
          </a>
        </li>
      ))}
    </ul>
  </div>
);

AlbumList.propTypes = {
  artistName: PropTypes.string,
  albums: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      albumArt: PropTypes.string,
      albumName: PropTypes.string,
      itunesLink: PropTypes.string
    })
  )
};

export default AlbumList;
