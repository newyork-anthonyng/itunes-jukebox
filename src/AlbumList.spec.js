import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import AlbumList from "./AlbumList";

const defaultProps = {
  artistName: "Michael Jackson",
  albums: [
    {
      albumArt:
        "https://is2-ssl.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/100x100bb.jpg",
      albumName: "The Essential Michael Jackson",
      artistName: "Michael Jackson",
      id: 159292399,
      itunesLink:
        "https://itunes.apple.com/us/album/the-essential-michael-jackson/159292399?uo=4"
    }
  ]
};

it("should render correctly", () => {
  const wrapper = shallow(<AlbumList {...defaultProps} />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("should render correctly when there are no albums", () => {
  const wrapper = shallow(<AlbumList {...defaultProps} albums={undefined} />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});
