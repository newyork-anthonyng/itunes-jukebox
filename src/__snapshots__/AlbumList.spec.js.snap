// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <h2>
    Michael Jackson
  </h2>
  <ul>
    <li
      key="159292399"
    >
      <a
        href="https://itunes.apple.com/us/album/the-essential-michael-jackson/159292399?uo=4"
        title="See The Essential Michael Jackson on iTunes"
      >
        <img
          alt="The Essential Michael Jackson"
          src="https://is2-ssl.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/100x100bb.jpg"
        />
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render correctly when there are no albums 1`] = `
<div>
  <h2>
    Michael Jackson
  </h2>
  <ul />
</div>
`;
